### ISSUE TITLE
Doesn't catch if the folder containing the app is moved

### ISSUE THREAD

**Issue by sarnau** [redsweater/RSAppMovementMonitor#6]
The monitor does not catch if the folder, which contains the running app, is moved e.g. into the trash.

**Comment by danielpunkass** [comment]
Thanks, I'll look into this!

**Comment by danielpunkass** [comment]
I reproduced it. What's interesting is at least in my tests, the movement is noticed and alerted but only after reactivating the app.

**Comment by danielpunkass** [comment]
Oh, hah! I guess this was a caveat I noticed originally: https://github.com/redsweater/RSAppMovementMonitor/blob/ad24b207c3dae57ff78d247885c162b96088ea24/RSAppMovementMonitor.swift#L80

Still I think I should be able to monitor more aggressively for this scenario. Will keep this ticket open until I have a chance to address this better.

**Comment by sarnau** [comment]
No worry, one of my QA guys noticed it. It works great when the app itself is moved (which is by far more common).

**Comment by sarnau** [comment]
I haven't had time to look into this myself so far (I need to take my vacation for this year…)

**Comment by danielpunkass** [comment]
That's cool! I'm glad your team is making use of it.

**Comment by sarnau** [comment]
I actually was a combination of cleaning up code, using a cleaner (= your) solution and also adding Swift to the code base (all ObjC/C++ so far) to test if that all works. We haven't shipped it yet.

**Comment by sarnau** [comment]
> Oh, hah! I guess this was a caveat I noticed originally:
> 
> https://github.com/redsweater/RSAppMovementMonitor/blob/ad24b207c3dae57ff78d247885c162b96088ea24/RSAppMovementMonitor.swift#L80
> 
> Still I think I should be able to monitor more aggressively for this scenario. Will keep this ticket open until I have a chance to address this better.

Yes, the app can load e.g. a NIB while being non-active. I think in our case the app was in the progress to become active again, but wasn't yet.

**Comment by danielpunkass** [comment]
Cool! I guess it might make sense for the movement monitor to check periodically e.g. every second whether it has moved, but I also was a little paranoid to be even that active.

**Comment by sarnau** [comment]
It will not be 100% perfect. What happens when you move the folder and within 1s activate the app again? At the end it is ok, it catches the most common issue of a crash (moving the app while running). I should file a bug against the Finder (to prohibit that) and maybe CFBundle (to correct the location of -mainBundle when the app is moved)

**Comment by danielpunkass** [comment]
Yeah. I may have neglected to actually file any bugs on this one ... but I noted some ways Apple could help a lot in the "Better Solutions" at the bottom of: https://redsweater.com/blog/3508/app-movement-monitoring

**Comment by sarnau** [comment]
I've added that to the bug reports, I've filed.

**Comment by danielpunkass** [comment]
Thanks!
